### PR TITLE
add support for SECONDARY_ENGINE

### PR DESCRIPTION
**Pull request by leiysky**
### What problem does this PR solve? 
MySQL syntax:
```
create_table_option:
        ......
        | SECONDARY_ENGINE_SYM opt_equal NULL_SYM
        | SECONDARY_ENGINE_SYM opt_equal ident_or_text
```
Issue: #405 

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -1631,6 +1632,14 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		} else {
 			ctx.WritePlainf("%d", n.UintValue)
 		}
+	case TableOptionSecondaryEngine:
+		ctx.WriteKeyWord("SECONDARY_ENGINE ")
+		ctx.WritePlainf("= ")
+		if n.StrValue == "NULL" {
```
Comment (kennytm) [inline ast/ddl.go]: `SECONDARY_ENGINE 'NULL'` and `SECONDARY_ENGINE NULL` are still semantically different, the former requests a secondary engine literally named "NULL".

File: ast/ddl.go [hunk 2]
```diff
@@ -1451,6 +1451,7 @@ const (
 	TableOptionIndexDirectory
 	TableOptionStorageMedia
 	TableOptionStatsSamplePages
+	TableOptionSecondaryEngine
```
Comment (leiysky) [inline ast/ddl.go]:
```suggestion
	TableOptionSecondaryEngine
	TableOptionSecondaryEngineNull
```

File: parser.y [hunk 3]
```diff
@@ -7661,6 +7662,22 @@ TableOption:
 		yylex.AppendError(yylex.Errorf("The STORAGE clause is parsed but ignored by all storage engines."))
 		parser.lastErrorAsWarn()
 	}
+|	"SECONDARY_ENGINE" EqOpt "NULL"
+	{
+		// Parse it but will ignore it
+		// See https://github.com/mysql/mysql-server/blob/8.0/sql/sql_yacc.yy#L5977-L5984
+		$$ = &ast.TableOption{Tp: ast.TableOptionSecondaryEngine, StrValue: "NULL"}
```
Comment (leiysky) [inline parser.y]:
```suggestion
		$$ = &ast.TableOption{Tp: ast.TableOptionSecondaryEngineNull}
```

File: ast/ddl.go (hunk 1 above)
Comment (leiysky) [inline ast/ddl.go]:
```suggestion
		ctx.WriteString(n.StrValue)
	case TableOptionSecondaryEngineNull:
		ctx.WriteKeyWord("SECONDARY_ENGINE ")
		ctx.WritePlain("= ")
		ctx.WriteKeyWord("NULL")
```

Comment (leiysky) [reply ast/ddl.go]: @kennytm What about this

Comment (kennytm) [reply ast/ddl.go]: @leiysky 👍

File: parser_test.go [hunk 4]
```diff
@@ -1740,6 +1740,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"create table t (c int) PACK_KEYS = DEFAULT", true, "CREATE TABLE `t` (`c` INT) PACK_KEYS = DEFAULT /* TableOptionPackKeys is not supported */ "},
 		{"create table t (c int) STORAGE DISK", true, "CREATE TABLE `t` (`c` INT) STORAGE DISK"},
 		{"create table t (c int) STORAGE MEMORY", true, "CREATE TABLE `t` (`c` INT) STORAGE MEMORY"},
+		{"create table t (c int) SECONDARY_ENGINE null", true, "CREATE TABLE `t` (`c` INT) SECONDARY_ENGINE = NULL"},
```
Comment (leoppro) [inline parser_test.go]: add a case:
```
create table t (c int) SECONDARY_ENGINE 'null'
```

Comment (leiysky) [reply parser_test.go]: added

**Comment by leoppro** [comment]
the integration test is failed, I'm working on it

**Comment by leoppro** [comment]
blocked by https://github.com/pingcap/tidb/pull/11553

**Comment by kennytm** [comment]
@leoppro The integration test isn't required here so it isn't really blocking any PRs.

**Comment by leoppro** [comment]
> @leoppro The integration test isn't required here so it isn't really blocking any PRs.

fine LGTM

**Comment by leoppro** [comment]
@kennytm PTAL

File: ast/ddl.go [hunk 5]
```diff
@@ -1631,6 +1633,14 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		} else {
 			ctx.WritePlainf("%d", n.UintValue)
 		}
+	case TableOptionSecondaryEngine:
+		ctx.WriteKeyWord("SECONDARY_ENGINE ")
+		ctx.WritePlainf("= ")
```
Comment (tangenta) [inline ast/ddl.go]:
```suggestion
		ctx.WritePlain("= ")
```

**Comment by leiysky** [comment]
@leoppro PTAL

**Pull request merged**